processing/src/test/java/org/apache/druid/query/UnionDataSourceTest.java

```diff
@@ -135,6 +136,9 @@ public void test_withChildren_sameNumber()
     );
   }
 
+  /*This test is failing for the cherry-pick adding aggregatorOverride to the UnioDataSource
+  * equals relies on aggregatorOverride, but hashCode does not. And the test is not available in 0.16 .*/
```

Comment:
to fix we can do like Objects.has(dataSources, aggregatorOverride), i already added that in my commit so you can ignore this for now

Reply:
aggregatorOverride is mutable in this case. so, if the value of aggregatorOverride changes, the same object may not be equal to itself at two different times. so, not sure fixing this test will satisfy the intent why this test was written in first place. As of now will wait to review your commit on it. Thanks.